File: frontend/src/types/crm.ts
export interface Lead {
  id: string;
  customer_name: string;
  phone_number?: string;
  project_name?: string;
  status: string;
  created_at: string;
  updated_at: string;
}

export interface PaginatedLeads {
  total: number;
  items: Lead[];
}

export interface Call {
  id: string;
  lead_id: string;
  livekit_call_id?: string;
  livekit_room_id?: string;
  vobiz_call_id?: string;
  recording_url?: string;
  transcript?: string; // May be omitted in list views
  summary?: string;
  duration_seconds?: number;
  status: string;
  created_at: string;
  // Joined fields
  customer_name?: string;
  phone_number?: string;
  project_name?: string;
}

export interface PaginatedCalls {
  total: number;
  items: Call[];
}

export interface Callback {
  id: string;
  lead_id: string;
  callback_requested: boolean;
  callback_date?: string;
  callback_time?: string;
  reason?: string;
  status: string;
  created_at: string;
  updated_at: string;
  // Joined fields
  customer_name?: string;
  phone_number?: string;
  project_name?: string;
}

export interface PaginatedCallbacks {
  total: number;
  items: Callback[];
}

File: backend/app/schemas/crm.py
from pydantic import BaseModel, Field
from typing import Optional
from uuid import UUID
from datetime import datetime

class LeadCreate(BaseModel):
    customer_name: str
    phone_number: Optional[str] = None
    project_name: Optional[str] = None
    status: Optional[str] = "NEW"

class LeadResponse(BaseModel):
    id: UUID
    customer_name: str
    phone_number: Optional[str] = None
    project_name: Optional[str] = None
    status: str
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class CallCreate(BaseModel):
    lead_id: UUID
    livekit_call_id: Optional[str] = None
    livekit_room_id: Optional[str] = None
    vobiz_call_id: Optional[str] = None
    recording_url: Optional[str] = None
    transcript: Optional[str] = None
    summary: Optional[str] = None
    duration_seconds: Optional[int] = None
    status: Optional[str] = "COMPLETED"

class CallResponse(BaseModel):
    id: UUID
    lead_id: UUID
    livekit_call_id: Optional[str] = None
    livekit_room_id: Optional[str] = None
    vobiz_call_id: Optional[str] = None
    recording_url: Optional[str] = None
    transcript: Optional[str] = None
    summary: Optional[str] = None
    duration_seconds: Optional[int] = None
    status: str
    created_at: datetime

    model_config = {"from_attributes": True}


class CallbackCreate(BaseModel):
    lead_id: UUID
    callback_requested: bool
    callback_date: Optional[str] = None
    callback_time: Optional[str] = None
    reason: Optional[str] = None
    status: Optional[str] = "PENDING"

class CallbackResponse(BaseModel):
    id: UUID
    lead_id: UUID
    callback_requested: bool
    callback_date: Optional[str] = None
    callback_time: Optional[str] = None
    reason: Optional[str] = None
    status: str
    created_at: datetime
    updated_at: datetime
    
    # Extra fields for list view joins
    customer_name: Optional[str] = None
    phone_number: Optional[str] = None
    project_name: Optional[str] = None

    model_config = {"from_attributes": True}

class LeadUpdate(BaseModel):
    status: Optional[str] = None

class CallbackUpdate(BaseModel):
    status: Optional[str] = None

# Pagination wrappers
class PaginatedLeads(BaseModel):
    total: int
    items: list[LeadResponse]

class CallListResponse(BaseModel):
    id: UUID
    lead_id: UUID
    livekit_call_id: Optional[str] = None
    livekit_room_id: Optional[str] = None
    vobiz_call_id: Optional[str] = None
    recording_url: Optional[str] = None
    summary: Optional[str] = None
    duration_seconds: Optional[int] = None
    status: str
    created_at: datetime
    
    # Extra fields for list view joins
    customer_name: Optional[str] = None
    phone_number: Optional[str] = None
    project_name: Optional[str] = None

    model_config = {"from_attributes": True}

class PaginatedCalls(BaseModel):
    total: int
    items: list[CallListResponse]

class PaginatedCallbacks(BaseModel):
    total: int
    items: list[CallbackResponse]
